### PR TITLE
Views

### PR DESCRIPTION
**Pull request by mahammad-mostafa**
- Implemented logic for both `index` & `show` methods
- Created views for (`users` / `user`  / `posts` / `post`)
- Built partials for all item types (`user` / `post` / `comment`)
- Styled all views with responsive layout 
- Added header & footer

### CODE REVIEW

File: app/views/posts/show.html.erb
Comment (aamirkhan2478) [inline app/views/posts/show.html.erb:1]: - [ ] Please show all comments in the `post show` view for now all comments are not shown on the show view so kindly fix it :+1: check below the comparison between your `post show view` and `wireframe show view` :slightly_smiling_face: 

**Your Design**
Screenshot from 2023-12-07 00-00-59

**Wireframe Design**
wireframe

Comment (aamirkhan2478) [inline app/views/posts/show.html.erb:1]: - [ ] Please make sure your design matches the wireframe but for now, your design does not match the wireframe you missed the `username` and also the `comment counter` and `like counter` should be on the right side so kindly fix this issue as well :slightly_smiling_face:

File: app/views/posts/index.html.erb
Comment (aamirkhan2478) [inline app/views/posts/index.html.erb:1]: - [ ] As I mentioned above :point_up:  please make sure your design matches with wireframe but your post's index page does not match with wireframe if you check your design some positions do not match for example the comment counter and like counter should be on the right side but in your case it's not on the right side so kindly fix issues as well :slightly_smiling_face: 

**Your design**
Screenshot from 2023-12-07 00-13-40

**Wireframe**
wireframe

File: app/views/users/index.html.erb
Comment (aamirkhan2478) [inline app/views/users/index.html.erb:1]: - [ ] As I mentioned above :point_up:  please make sure your design matches with wireframe but your user's index page does not match with wireframe if you check your design some positions do not match for example the post counter should be on the right side but in your case it's not on the right side so kindly fix issues as well :slightly_smiling_face: 

**Your design**
Screenshot from 2023-12-07 00-16-02

**Wireframe**
wireframe

File: app/views/users/show.html.erb
Comment (aamirkhan2478) [inline app/views/users/show.html.erb:1]: - [ ] As I mentioned above :point_up:  please make sure your design matches with wireframe but your user's show page does not match with wireframe if you check your design some positions do not match for example the comment counter and like counter and post counter should be on the right side but in your case it's not on the right side so kindly fix issues as well :slightly_smiling_face: 

**Your design**
Screenshot from 2023-12-07 00-18-48


**Wireframe**
wireframe

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @mahammad-mostafa, 



Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Good Effort
✔️ Your design is good.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job making the requested changes.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**